File: componentbase/src/main/java/com/learn/componentbase/service/IAccountService.java
package com.learn.componentbase.service;

import android.app.Activity;
import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;

/**
 * @Author: LL
 * @Description:
 * @Date:Create：in 2021/1/15 14:12
 */
public interface IAccountService {
    /**
     * 是否已经登录
     *
     * @return
     */
    boolean isLogin();

    /**
     * 获取登录用户的 AccountId
     *
     * @return
     */
    String getAccountId();


    /**
     * 创建 UserFragment
     *
     * @param activity
     * @param containerId
     * @param manager
     * @param bundle
     * @param tag
     * @return
     */
    Fragment newUserFragment(Activity activity, int containerId, FragmentManager manager, Bundle bundle, String tag);
}

File: export_account/src/main/java/com/wyt/export_account/service/IAccountService.kt
package com.wyt.export_account.service

import android.app.Activity
import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import com.alibaba.android.arouter.facade.template.IProvider
import com.wyt.export_account.bean.UserInfo

/**
 * @Author: LL
 * @Description:
 * @Date:Create：in  2021/1/29 15:14
 */
interface IAccountService : IProvider {
    /**
     * 是否已经登录
     *
     * @return
     */
    fun isLogin(): Boolean

    /**
     * 获取登录用户的 AccountId
     *
     * @return
     */
    fun getAccountId(): String?

    /**
     * 设置userinfo
     */
    fun setUserInfo(userInfo: UserInfo?)

    /**
     * 获取userinfo
     */
    fun getUserInfo(): UserInfo


    /**
     * 创建 UserFragment
     *
     * @param activity
     * @param containerId
     * @param manager
     * @param bundle
     * @param tag
     * @return
     */
    fun newUserFragment(activity: Activity?, containerId: Int, manager: FragmentManager?, bundle: Bundle?, tag: String?): Fragment?
}